app/libs/stats.py

```diff
@@ -1,33 +1,575 @@
+            fn.count(Comment.id)
+        ).group_by(
+            Comment.commenter
+        ).where(
+            Comment.editors_pick == True
+        ).tuples()
+    count = sum([count for commenter, count in featured_comments])
+    commenters = [(commenter['name'], count)
+                  for commenter, count in featured_comments]
+    return {'count': count, 'commenters': commenters}
+get_featured_comments.groups_required = [groups.moderator.value]
+
+
+# returns a dict containing rejected comments count and their commenters categorized by reason
+def get_rejected_comments(since=None):
+    rejected_comments = {i.name: {} for i in rejection_reasons}
+    rejected = RejectedComment.select(
+            RejectedComment.reason,
+            RejectedComment.commenter,
+            fn.count(RejectedComment.id)
+        ).group_by(
+            RejectedComment.reason,
+            RejectedComment.commenter
+        )
+    if since:
+        rejected = rejected.where(RejectedComment.created >= since)
+    for reasons in rejected_comments.keys():
+        count = sum([
+            count for reason, commenter, count in rejected.tuples()
+            if reason == rejection_reasons[reasons].value
+        ])
+        commenters = [(commenter['name'], count)
+                      for reason, commenter, count in rejected.tuples()
+                      if reason == rejection_reasons[reasons].value]
+        rejected_comments[reasons]['count'] = count
+        rejected_comments[reasons]['commenter'] = commenters
+    return rejected_comments
+get_rejected_comments.groups_required = [groups.moderator.value]
+
+
+def get_rejected_comments_lastNmonths(n=4):
+    return get_rejected_comments(
+        arrow.utcnow().shift(months=-int(n)).span('month')[0].date())
+get_rejected_comments_lastNmonths.groups_required = [groups.moderator.value]
+
+
+def get_week_or_month_counter(metric, first_metric_value, last_metric_value):
+    metric_counter = first_metric_value
+    metric_values = []
+    while (metric_counter <= last_metric_value):
+        metric_values.append(metric_counter)
+        if metric == 'month':
+            days_in_curr_metric_counter = calendar.monthrange(
+                metric_counter.year, metric_counter.month)[1]
+            metric_counter += datetime.timedelta(
+                days=days_in_curr_metric_counter)
+        else:
+            metric_counter += datetime.timedelta(weeks=1)
+    return metric_values
+
+
+def merge_approved_rejected_list(metric_counter, approved, rejected):
+    app_counter = 0
+    rej_counter = 0
+    total_comments = []
+    for value in metric_counter:
+        if (app_counter < len(approved) and approved[app_counter][0] == value
+            ) and (rej_counter < len(rejected)
+                   and rejected[rej_counter][0] == value):
+            m1, c = approved[app_counter]
+            m2, rc = rejected[rej_counter]
+            total_comments.append((m1.date().isoformat(), c, rc, (c + rc),
```

Comment:
@pramitrn please use proper variable names. `m1, c, m2, rc` are not easy to understand.